clash-prelude/src/Clash/Sized/Vector.hs

```diff
@@ -1687,18 +1686,24 @@ at n xs = head $ snd $ splitAt n xs
 -- 2 :> 4 :> 6 :> Nil
 -- >>> select d1 d2 d3 (1:>2:>3:>4:>5:>6:>7:>8:>Nil)
 -- 2 :> 4 :> 6 :> Nil
-select :: (CmpNat (i + s) (s * n) ~ 'GT)
+select :: forall i s n f a. s * n + 1 <= i + s
```

Comment:
I think this constitutes an API change. The order of the tyvars is unchanged, but the constraint has changed. API changes should get a Changelog entry.

If we look at the origin of the uncommon constraint that this had, it's [this commit](https://github.com/clash-lang/clash-compiler/commit/461e14982a0c85bd22b5d9358fb67a3ec1ed8033). It seems to me the suggested change in this PR is an improvement that satisfies the exact same constraint, unless I'm missing some shortcoming in our solver plugins... however, the solver can't prove the equality of the old and new constraints, so users might have to rephrase the constraints in their designs.

Reply:
I added a corresponding changelog entry for it.